### PR TITLE
YARP Plugin. AzureAD Plugin improvements

### PR DESCRIPTION
**Pull request by Exmirai**

### CODE REVIEW

File: plugins/OnlineSales.Plugin.AzureAD/AzureADPlugin.cs [hunk 1]
```diff
@@ -2,24 +2,88 @@
...
+
 public class AzureADPlugin : IPlugin, ISwaggerConfigurator, IPluginApplication
 {
     public void ConfigureServices(IServiceCollection services, IConfiguration configuration)
     {
         var administratorsGroupId = configuration.GetValue<string>("AzureAd:GroupsMapping:Administrators");
 
-        services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
-                    .AddMicrosoftIdentityWebApi(configuration, subscribeToJwtBearerMiddlewareDiagnosticsEvents: true);
+        services.AddAuthentication("ApiAppAuthorization")
```
Comment (idchrono) [inline plugins/OnlineSales.Plugin.AzureAD/AzureADPlugin.cs:36]: move authentification code to separate file

File: plugins/OnlineSales.Plugin.AzureAD/Controllers/AuthController.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+[Route("api/auth")]
+public class AuthController : ControllerBase
+{
+    public AuthController()
+    {
+    }
+
+    [HttpGet("login")]
+    public async Task Login(string redirectUrl)
```
Comment (idchrono) [inline plugins/OnlineSales.Plugin.AzureAD/Controllers/AuthController.cs:20]: describe possible responses for swagger

File: plugins/OnlineSales.Plugin.ReverseProxy/ReverseProxyPlugin.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        var proxyConfig = configuration.GetSection("YARPSettings");
+        if (proxyConfig == null)
+        {
+            throw new MissingConfigurationException("YARP configuration is mandatory.");
+        }
+
+        services.AddAuthorization(options =>
+        {
+            options.AddPolicy("ProxyAuth", policy =>
+                policy.RequireAuthenticatedUser().AddAuthenticationSchemes("ApiAppAuthorization"));
```
Comment (idchrono) [inline plugins/OnlineSales.Plugin.ReverseProxy/ReverseProxyPlugin.cs:26]: you use "ApiAppAuthorization" freqrently enough to move it to constant

**Review by idchrono — CHANGES_REQUESTED** [review]
check comments above

File: plugins/OnlineSales.Plugin.AzureAD/AzureADPlugin.cs [hunk 4]
```diff
@@ -2,24 +2,33 @@
...
+using Microsoft.AspNetCore.Http;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
-using Microsoft.Identity.Web;
 using Microsoft.OpenApi.Models;
 using Swashbuckle.AspNetCore.SwaggerGen;
 
 namespace OnlineSales.Plugin.AzureAD;
 
+public class JwtBearerEventsHandler : JwtBearerEvents
```
Comment (idchrono) [inline plugins/OnlineSales.Plugin.AzureAD/AzureADPlugin.cs:16]: move "JwtBearEventHandler" out to separate file

File: tests/OnlineSales.Tests/Environment/TestApplication.cs [hunk 5]
```diff
@@ -73,9 +73,9 @@ protected override IHost CreateHost(IHostBuilder builder)
             services.AddScoped<IEmailValidationExternalService, TestEmailValidationExternalService>();
             services.AddScoped<IAccountExternalService, TestAccountExternalService>();
 
-            services.AddAuthentication(defaultScheme: "TestScheme")
+            services.AddAuthentication(defaultScheme: "WebApiAuthorization")
```
Comment (idchrono) [inline tests/OnlineSales.Tests/Environment/TestApplication.cs:76]: make constant "WebApiAuthorization"

**Review by idchrono — CHANGES_REQUESTED** [review]
check comments above

File: plugins/OnlineSales.Plugin.AzureAD/AzureADPlugin.cs [hunk 6]
```diff
@@ -23,9 +24,15 @@ public void ConfigureServices(IServiceCollection services, IConfiguration config
         {
             options.AddPolicy("Administrators", opts =>
             {
-                opts.RequireClaim(
-                    "http://schemas.microsoft.com/ws/2008/06/identity/claims/role",
-                    administratorsGroupId ?? Guid.NewGuid().ToString());
+                if (!string.IsNullOrEmpty(administratorsGroupId))
```
Comment (idchrono) [inline plugins/OnlineSales.Plugin.AzureAD/AzureADPlugin.cs:27]: please change to if null throw exception

**Review by idchrono — CHANGES_REQUESTED** [review]
see comments above

**Review by idchrono — APPROVED** [review]

File: plugins/OnlineSales.Plugin.AzureAD/AzureADPlugin.cs [hunk 7]
```diff
@@ -18,20 +18,30 @@ public void ConfigureServices(IServiceCollection services, IConfiguration config
...
+        {
+            options.MinimumSameSitePolicy = SameSiteMode.None;
+            options.Secure = CookieSecurePolicy.Always;
+        });
     }
-    
+
     public void ConfigureSwagger(SwaggerGenOptions options, OpenApiInfo settings)
     {
         options.AddSecurityDefinition("Bearer", new OpenApiSecurityScheme()
```
Comment (peterliapin) [inline plugins/OnlineSales.Plugin.AzureAD/AzureADPlugin.cs:46]: @Exmirai new cookie based authentication should also be supported in swagger UI. Add it here.

**Pull request merged**